x-pack/plugins/apm/public/components/app/TraceLink/get_redirect_to_transaction_detail_page_url.ts

```diff
@@ -14,15 +14,48 @@ export const getRedirectToTransactionDetailPageUrl = ({
   transaction: Transaction;
   rangeFrom?: string;
   rangeTo?: string;
-}) =>
-  format({
+}) => {
+  return format({
     pathname: `/services/${transaction.service.name}/transactions/view`,
     query: {
       traceId: transaction.trace.id,
       transactionId: transaction.transaction.id,
       transactionName: transaction.transaction.name,
       transactionType: transaction.transaction.type,
-      rangeFrom,
-      rangeTo,
+      rangeFrom:
+        rangeFrom ||
+        roundToNearestMinute({
+          timestamp: transaction['@timestamp'],
+          direction: 'down',
+        }),
+      rangeTo:
+        rangeTo ||
+        roundToNearestMinute({
+          timestamp: transaction['@timestamp'],
+          diff: transaction.transaction.duration.us / 1000,
+          direction: 'up',
+        }),
     },
   });
+};
+
+export function roundToNearestMinute({
```

Comment:
Is this only exported for the purpose of testing? Would it be better to test this indirectly by testing `getRedirectToTransactionDetailPageUrl` in order to reduce the exported API surface area?

Reply:
Addressed in https://github.com/elastic/kibana/pull/80993/commits/80dc9ec82ab6aa95489fffbedd3d631ee68ee2c8